orttraining/orttraining/training_ops/cuda/communication/nccl_service.cc

```diff
@@ -0,0 +1,346 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#ifdef USE_NCCL
+
+#include "core/common/common.h"
+#include "core/profile/context.h"
+#include "core/providers/cuda/cuda_common.h"
+#include "orttraining/core/framework/mpi_context.h"
+#include "orttraining/training_ops/cuda/communication/nccl_service.h"
+#include <iostream>
+#include <nccl.h>
+
+namespace onnxruntime {
+namespace cuda {
+
+bool NcclTask::Compare(const NcclTask& other) const {
+  if (type != other.type) {
+    return false;
+  }
+  if (peers.size() != other.peers.size()) {
+    return false;
+  }
+  for (size_t i = 0; i < peers.size(); ++i) {
+    if (peers[i] != other.peers[i]) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+void NcclTask::ResetTask() {
```

Comment:
>ResetTask [](start = 15, length = 9)

Is this function being used? I guess it is not completely 'Reset' the task, instead, it clears out some of its fields but leave peers and type intact for future re-use?